spec/go-spec.coffee

```diff
@@ -106,32 +106,45 @@ describe 'Go grammar', ->
       expect(tokens[0].value).toEqual '\'' + verb + '\'',
       expect(tokens[0].scopes).toEqual ['source.go', 'constant.other.rune.go']
 
-  it 'tokenizes invalid runes and single quote strings', ->
+  it 'tokenizes invalid runes and single quoted strings', ->
     {tokens} = grammar.tokenizeLine('\'ab\'')
     expect(tokens[0].value).toEqual '\'ab\''
-    expect(tokens[0].scopes).toEqual ['source.go', 'invalid.illegal.rune.go']
+    expect(tokens[0].scopes).toEqual ['source.go', 'invalid.illegal.unknown-rune.go']
 
     {tokens} = grammar.tokenizeLine('\'some single quote string\'')
     expect(tokens[0].value).toEqual '\'some single quote string\''
-    expect(tokens[0].scopes).toEqual ['source.go', 'invalid.illegal.rune.go']
+    expect(tokens[0].scopes).toEqual ['source.go', 'invalid.illegal.unknown-rune.go']
 
   it 'tokenizes invalid whitespace around chan annotations', ->
-    invalids =
+    invalid_send =
       'chan <- sendonly': ' '
+
+    invalid_receive =
       '<- chan recvonly': ' '
 
-    for expr, invalid of invalids
+    for expr, invalid of invalid_send
+      {tokens} = grammar.tokenizeLine(expr)
+      expect(tokens[1].value).toEqual invalid
+      expect(tokens[1].scopes).toEqual ['source.go', 'invalid.illegal.send-channel.go']
+
+    for expr, invalid of invalid_receive
       {tokens} = grammar.tokenizeLine(expr)
       expect(tokens[1].value).toEqual invalid
-      expect(tokens[1].scopes).toEqual ['source.go', 'invalid.illegal.go']
+      expect(tokens[1].scopes).toEqual ['source.go', 'invalid.illegal.receive-channel.go']
 
   it 'tokenizes keywords', ->
     keywordLists =
-      'keyword.go': ['func', 'var', 'const', 'type', 'struct', 'interface', 'case', 'default']
-      'keyword.directive.go': ['package', 'import']
-      'keyword.statement.go': ['defer', 'go', 'goto', 'return', 'break', 'continue', 'fallthrough']
-      'keyword.conditional.go': ['if', 'else', 'switch', 'select']
-      'keyword.repeat.go': ['for', 'range']
+      'keyword.control.go': ['break', 'case', 'continue', 'default', 'defer', 'else', 'fallthrough', 'for', 'go', 'goto', 'if', 'range', 'return', 'select', 'switch']
+      'keyword.channel.go': ['chan']
+      'keyword.constant.go': ['const']
+      'keyword.function.go': ['func']
+      'keyword.interface.go': ['interface']
+      'keyword.import.go': ['import']
+      'keyword.map.go': ['map']
+      'keyword.package.go': ['package']
+      'keyword.struct.go': ['struct']
+      'keyword.type.go': ['type']
+      'keyword.variable.go': ['var']
```

Comment:
Sub-scope should be `var` instead of `variable` because it match "var" symbol. `var` is not a variable. In addition it will not cause misunderstanding because of `variable` root group.